src/plugins/discover/public/application/helpers/get_sharing_data.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { IUiSettingsClient } from 'kibana/public';
+import { DOC_HIDE_TIME_COLUMN_SETTING, SORT_DEFAULT_ORDER_SETTING } from '../../../common';
+import { getSortForSearchSource } from '../angular/doc_table';
+import { SearchSource } from '../../../../data/common';
+import { AppState } from '../angular/discover_state';
+import { SortOrder } from '../../saved_searches/types';
+
+const getSharingDataFields = async (
+  getFieldCounts: () => Promise<Record<string, number>>,
+  selectedFields: string[],
+  timeFieldName: string,
+  hideTimeColumn: boolean
+) => {
+  if (selectedFields.length === 1 && selectedFields[0] === '_source') {
+    const fieldCounts = await getFieldCounts();
+    return {
+      searchFields: undefined,
+      selectFields: Object.keys(fieldCounts).sort(),
+    };
+  }
+
+  const fields =
+    timeFieldName && !hideTimeColumn ? [timeFieldName, ...selectedFields] : selectedFields;
+  return {
+    searchFields: fields,
+    selectFields: fields,
+  };
+};
+
+/**
+ * Preparing data to share the current state as link or CSV/Report
+ */
+export async function getSharingData(
+  currentSearchSource: SearchSource,
+  state: AppState,
```

Comment:
Can we avoid passing in the entire app state here? Seems like we only need sort parameter.

Reply:
we'd need`sort` and `sort`, so if we would use both as separate arguments, we would have 5 arguments, if there was a new state param we would like to add it would be 6. In this case I'd prefer to read the function in a way like "Take search source and use app state and a bit more to update it"  ... and get rid of `getFieldCounts` soon :)